warehouse/core/src/main/java/datawave/ingest/mapreduce/job/SplittableRFileInputFormat.java

```diff
@@ -0,0 +1,137 @@
+    public RecordReader<Key,Value> createRecordReader(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+        return new SplittableRFileRecordReader();
+    }
+
+    /**
+     * Create rfile splits for the configured job. See <code>FileInputFormat</code>
+     *
+     * @param job
+     * @return
+     * @throws IOException
+     */
+    @Override
+    public List<InputSplit> getSplits(JobContext job) throws IOException {
+        Configuration config = job.getConfiguration();
+
+        List<InputSplit> rfileSplits = new ArrayList<>();
+        log.info("getting splits for job");
+        // get the configured directories/files
+        for (InputSplit inputSplit : super.getSplits(job)) {
+            if (!(inputSplit instanceof FileSplit)) {
+                throw new IllegalArgumentException("Must have file splits");
+            }
+
+            // for each file get the index blocks
+            FileSplit fileSplit = (FileSplit) inputSplit;
+            log.info("creating splits for: " + fileSplit);
+            rfileSplits.addAll(getSplits(config, fileSplit));
+        }
+        log.info("total splits: " + rfileSplits.size());
+
+        return rfileSplits;
+    }
+
+    /**
+     * Create rfile splits for a given FileSplit. Splits will honor the MIN_BLOCKS_PER_SPLIT config option. Splits may extend more than MIN_BLOCKS_PER_SPLIT if
+     * a given Key spans more index blocks.
+     *
+     * @param config
+     * @param fileSplit
+     * @return
+     * @throws IOException
+     */
+    public static List<InputSplit> getSplits(Configuration config, FileSplit fileSplit) throws IOException {
+        log.info("getting splits for: " + fileSplit);
+        Path rfile = fileSplit.getPath();
+        FileSystem fs = rfile.getFileSystem(config);
+
+        if (!fs.exists(rfile)) {
+            throw new IllegalArgumentException(rfile + " does not exist");
+        }
+
+        CryptoService cs = CryptoFactoryLoader.getServiceForClient(CryptoEnvironment.Scope.TABLE, config.getPropsWithPrefix(TABLE_CRYPTO_PREFIX.name()));
+        CachableBlockFile.CachableBuilder cb = new CachableBlockFile.CachableBuilder().fsPath(fs, rfile).conf(config).cryptoService(cs);
+        RFile.Reader rfileReader = new RFile.Reader(cb);
+
+        // get the first and last keys to bound the blocks while creating splits
+        Key firstKey = rfileReader.getFirstKey();
+        Key lastKey = rfileReader.getLastKey();
+
+        // use the index blocks to create the splits
+        FileSKVIterator iter = rfileReader.getIndex();
+
+        // track the last split key to since multiple splits with the same split key MUST be in the same block
+        Key lastSplit = firstKey;
+        List<InputSplit> splits = new ArrayList<>();
+        int blkCount = 0;
+        int splitBlocks = 0;
+
+        int minBlocksPerSplit = config.getInt(MIN_BLOCKS_PER_SPLIT, 1);
+        log.info("Blocks per split: " + minBlocksPerSplit);
```

Comment:
This is going to log the same information for every underlying RFile we process. I'm picturing the log looking like:
```
[INFO] Blocks per split: 100
[INFO] Blocks per split: 100
...
```
Might be worth bumping this out to the top level `getSplits()`

Reply:
agree